extensions/vscode/src/state.ts

```diff
@@ -347,6 +359,7 @@ export class PublisherState implements Disposable {
 
       const listResponse = await api.credentials.list();
       this.credentials = listResponse.data;
+      this.credentialRefresh.fire({ oldCredentials: oldCredentials });
```

Comment:
I really like the switch here over to firing an event. We used to have a specialized event bus for our various components spread out within VSCode, but as the Publisher view became more monolithic, it was removed to simplify the codebase. You solved your problem of notifying the authentication component by leveraging VSCode's facility, which looks to work nicely.